narwhals/_arrow/series.py

```diff
@@ -568,14 +568,73 @@ def sample(
 
         return self._from_native_series(pc.take(ser, mask))
 
-    def fill_null(self: Self, value: Any) -> Self:
+    def fill_null(
+        self: Self,
+        value: Any | None = None,
+        strategy: Literal["forward", "backward"] | None = None,
+        limit: int | None = None,
+    ) -> Self:
+        if value is not None and strategy is not None:
+            msg = "cannot specify both `value` and `strategy`"
+            raise ValueError(msg)
+        if value is None and strategy is None:
+            msg = "must specify either a fill `value` or `strategy`"
+            raise ValueError(msg)
+        if strategy is not None and strategy not in {"forward", "backward"}:
+            msg = f"strategy not supported: {strategy}"
+            raise ValueError(msg)
+
+        import numpy as np  # ignore-banned-import
         import pyarrow as pa  # ignore-banned-import()
         import pyarrow.compute as pc  # ignore-banned-import()
 
+        def fill_aux(
```

Comment:
Big kudos 🙌🏼 this is ingenious - I like it a lot.
Since it took me 5-10 mins and writing the code to see what happens, would you mind adding two lines with the high level idea for our future selves? 😇

Reply:
Thank you! I had to make a few adjustments to pass the min and old versions test. 

I've added the comments as requested c: this will definitely help our future selves!